src/pymovements/dataset/resources.py

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) 2025 The pymovements Project Authors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Resources and Resource module."""
+from __future__ import annotations
+
+from collections.abc import Iterable
+from collections.abc import Sequence
+from copy import deepcopy
+from dataclasses import asdict
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass
+class Resource:
+    """Resource definition."""
+
+    content: str
+    filename: str | None = None
```

Comment:
I'm unsure about this field name. #1162 will add `Resource.filename_pattern` which may be confusing:

- `Resource.filename` is the filename of the downloaded source file.
- `Resource.filename_pattern` is the pattern of the single files (these can be members of archives)

So maybe the field should be renamed to something like `source_filename`? But I would be in favor to leave it as is for this PR and improve later on when improving and extending the `Resource` class.